msbuild/Xamarin.Shared/Xamarin.Shared.props

```diff
@@ -264,7 +264,8 @@ Copyright (C) 2020 Microsoft. All rights reserved.
 
 	<!-- This is used to determine if VS is connected to mac. -->
 	<PropertyGroup>
-		<IsMacEnabled>true</IsMacEnabled>
+		<IsMacEnabled Condition="'$(IsMacEnabled)' == '' And '$(OS)' != 'Windows_NT'">true</IsMacEnabled>
+		<IsMacEnabled Condition="'$(IsMacEnabled)' == '' And '$(OS)' == 'Windows_NT'">false</IsMacEnabled>
```

Comment:
Whoops, I remembered how a user had the `%OS%` environment variable set to `Windows` 🤦 and it breaks this check. It was set on their Dell PC out of the box...

We've been doing this as the new way to check the OS in MSBuild:
* `Condition="$([MSBuild]::IsOSPlatform('windows'))"`
* `Condition="$([MSBuild]::IsOSPlatform('osx'))"`
* `Condition="$([MSBuild]::IsOSPlatform('linux'))"`

Reply:
I guess the strings here map to [OSPlatform](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.osplatform?view=netcore-3.1).

https://github.com/dotnet/msbuild/blob/295bf2e8355b6e8e3454a3a3bb4511d255455506/src/Build/Evaluation/IntrinsicFunctions.cs#L439-L442

Reply:
@jonathanpeppers I fixed it to use `$([MSBuild]::IsOSPlatform('windows'))`. Good catch!